engine/gamesys/src/gamesys/gamesys_private.cpp

```diff
@@ -528,23 +528,20 @@ namespace dmGameSystem
         // Otherwise, we try to get it from the component itself via the callabck.
         // We need a callback mechanism here because we don't have a shared data layout for
         // the override attributes on component level (TODO: we should have).
-        else
```

Comment:
This was the culprit. If we do have an overridden value index but the exact attribute we are looking for has _not_ been overridden, we would never actually get the data for it..